spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkDataWrite.java

```diff
@@ -71,9 +72,18 @@ public class TestSparkDataWrite {
 
   @Rule public TemporaryFolder temp = new TemporaryFolder();
 
-  @Parameterized.Parameters(name = "format = {0}")
+  private String targetBranch;
+
+  @Parameterized.Parameters(name = "format = {0}, branch = {1}")
   public static Object[] parameters() {
-    return new Object[] {"parquet", "avro", "orc"};
+    return new Object[][] {
+      new Object[] {"parquet", "main"},
+      new Object[] {"parquet", "testBranch"},
```

Comment:
We don't need to test all formats and branch/main combinations. We can just do parquet/main, parquet/testBranch, avro/main, orc/testBranch.